src/Cli/dotnet/commands/dotnet-msbuild/MSBuildForwardingApp.cs

```diff
@@ -40,6 +41,25 @@ private static IEnumerable<string> ConcatTelemetryLogger(IEnumerable<string> arg
 
         public MSBuildForwardingApp(IEnumerable<string> argsToForward, string msbuildPath = null)
         {
+            var foundDir = false;
+            foreach (var arg in argsToForward)
```

Comment:
The core question I have is - why wouldn't we try to fix this in MSBuild itself? I could imagine a new CLI flag to `--fail-on-root-drive-enumeration` that the CLI could pass, and that would allow MSBuild to accurately parse and check the problem once, for all callers. 

cc @rainersigwald @ladipro for thoughts? The core problem we're trying to fix is that enumerating a path glob from drive root (which is common with SDK-style projects and containers) takes _forever_ in MSBuild and you only get a warning after it's happened, if at all. It would be good to have a stricter mode here.

Reply:
> you only get a warning after it's happened, if at all

We're fixing that: https://github.com/dotnet/msbuild/issues/9818.

There's an existing opt-into-quick-failure for this case: `MSBUILDFAILONDRIVEENUMERATINGWILDCARD=1`. I wouldn't be super opposed to setting this in the SDK.

Containerized build environments are a potential case where someone might intentionally have a project at the root, but even then it's not great--you still need the SDK and enough tools to use it inside the container so it's best to (and our samples do) copy your code into `/build/` or similar.

Reply:
Oh excellent - @Forgind can you trial using this flag? The idea would be we'd always set this environment variable unless a user passed a `--force-enumeration` (or equivalent) SDK flag.

Reply:
That sounds 1000% better than my ugly hacky thing, thanks! Will try it out and see what happens.

Reply:
I tried out MSBUILDFAILONDRIVEENUMERATINGWILDCARD=1, and it worked like a charm! Well, at least in the sense that it correctly detected the drive enumeration and through an error. It pointed to a specific Item Include, and I think a naive user might have no idea why that MSBuild code executed, but if they read a bit further, it says something about a wildcard that results in enumerating all files on the drive, so I call that a win?

```
C:\Program Files\dotnet\sdk\8.0.200\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.DefaultItems.props(30,14): error M
SB5029: The value "**/*.cs" of the "Include" attribute in element <ItemGroup> in file "C:\Program Files\dotnet\sdk\8.0.
200\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.DefaultItems.props (30,14)" is a wildcard that results in enumerat
ing all files on the drive, which was likely not intended. Check that referenced properties are always defined. [C:\C.c
sproj]
```

@baronfel, what would you think of setting that unconditionally if it isn't already set (i.e., instead of --force-enumeration, have the user set MSBUILDFAILONDRIVEENUMERATINGWILDCARD=0)? --force-enumeration is probably fine but a bit more difficult.

Reply:
I'm fine with setting that unconditionally if it isn't already set - we can document the workaround of the environment variable. From a user experience point of view, I'd _love_ to see if we could massage that message a bit - can the CLI tell the user more forcefully when this error occurs to stop doing the bad thing of building in the drive root? Can you drop a screenshot of what this looks like in-context, both in the console logger and in terminal logger?

Reply:
Sorry this is underwhelming, but this is literally all there is in my console:
<img width="673" alt="image" src="https://github.com/dotnet/sdk/assets/12969783/85675457-aafd-4006-aee0-8c1a6a159735">

Reply:
After a brief flash (too fast for me to tell what was happening), tl gave me this:
<img width="673" alt="image" src="https://github.com/dotnet/sdk/assets/12969783/4ada6d69-a8fa-434c-b77f-cd8fbe2387b2">

Reply:
@Forgind I was thinking similarly - the CLI has a logger that it adds to builds - maybe we could detect that build-failed condition and emit something to the stdout after the build has failed that's more...clean to a user?